fvm/src/kernel/default.rs

```diff
@@ -366,6 +366,10 @@ where
         self.caller
     }
 
+    fn msg_origin(&self) -> Result<Option<ActorID>> {
+        self.resolve_address(&self.call_manager.origin())
```

Comment:
This method charges gas. We should pre-resolve origin when we construct the call manager. Really, we just need to pass `sender_id` instead of `msg.from` in https://github.com/filecoin-project/ref-fvm/blob/ef808624f20611fb0e6079d9c8d24193d4f4d440/fvm/src/executor/default.rs#L69.

Reply:
ack, will do.

Reply:
Hm, it doesn't? There is no syscall gas (we already charged it), and the kernel method doesn't charge explicit gas. However, I'm also fine with pre-resolving when constructing the call manager (would also involve less work when origin is called more than once in the call stack).

Reply:
Regardless: I agree with @Stebalien. There's no reason for the origin in the CallManager to be an address. It can just be an ActorID and problem solved. Implemented here: https://github.com/filecoin-project/ref-fvm/commit/bfb70ef6e2f1f4a5199d070c61f2eac5a89a2e7f